README.md

```diff
@@ -30,6 +30,54 @@ Any of the extensions listed below can be mixed into a class by `include`-ing `H
 
 ### Coercion
 
+Three systems are available for coercion: ActiveModel types, dry-types, and hashie's builtin types. The use of a specified coercion system is determined by exactly which Coercion module you include in your Hashie class. Including the default `Hashie::Extensions::Coercion` module will use Hashie's builtin type system.
+
+### Coercion using ActiveModel types
+
+To use ActiveModel's type coercion system, you must add version 5 of ActiveModel to your Gemfile. If you are using Rails, activemodel is included, but you must be using Rail's version 5 in order to use this. If you are on running a lower version of Rails, you will not be able to use ActiveModel's type coercion system due to Ruby's inability to use different Gem versions in the same application.
+
+```ruby
+gem 'activemodel', '~> 5.x'
+```
+
+Then include the `active_model` Coercion module into your Hashie class:
+
+```ruby
+class Tweet < Hash
+  include Hashie::Extensions::Coercion.active_model
```

Comment:
**Problems with dynamic modules**

The problem I see with doing anonymous module building is that when you're trying to debug an issue, it gets harder to look at where a method comes from when it shows up in `Stuff.ancestors` as a `<Module#...>` rather than `<Hashie::Extensions::Coercion::ActiveModel>`.

I thought the dynamic module idea was interesting a while ago (in fact, I [have an in progress feature for Hashie](https://github.com/michaelherold/hashie/tree/persistence-framework) using a [framework I built for building modules](https://github.com/michaelherold/module_builder) that uses this technique) but ran up against that problem.

Do we see that as an issue?

**Reason for approach**

Secondly, is the reason for this approach just to eliminate a little bit of duplicate code? I feel like it can actually make it harder for follow what's going on in the source, which makes maintenance harder down the road. If it's just for that purpose, I think it would be better to just be explicit and not rely on metaprogramming.

If there's a more in-depth reason, I'm all ears!